[functorch] fix conv_transpose with groups batching rule

zou3519 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #84939
* __->__ #84938

The original batching rule didn't work in all cases, so this PR fixes it.

Test Plan:
 - added new test case to conv_transpose2d's OpInfo. Surprisingly the
 other test cases didn't catch the bug.
 - Removed some xfails and skips as a result.

zou3519 committed:
[functorch] fix conv_transpose with groups batching rule

The original batching rule didn't work in all cases, so this PR fixes it.

Test Plan:
 - added new test case to conv_transpose2d's OpInfo. Surprisingly the
 other test cases didn't catch the bug.
 - Removed some xfails and skips as a result.

[ghstack-poisoned]

samdow commented on functorch/functorch/csrc/BatchRulesConvolution.cpp:
```diff
@@ -52,15 +52,47 @@ convolution_batch_rule(const Tensor& lhs, optional<int64_t> lhs_bdim, const Tens
+        // N(GBO)HW -> NG(BO)HW
+        out = reshape_dim_outof(1, groups, out);
+        // NG(BO)HW -> NGBOHW
+        out = reshape_dim_outof(2, rhs.sizes()[*rhs_bdim], out);
```
nit: could slightly simplify as `rhs.size(*rhs_bdim)`

samdow commented on functorch/functorch/csrc/BatchRulesConvolution.cpp:
```diff
@@ -52,15 +52,47 @@ convolution_batch_rule(const Tensor& lhs, optional<int64_t> lhs_bdim, const Tens
+        // N(GBO)HW -> NG(BO)HW
+        out = reshape_dim_outof(1, groups, out);
+        // NG(BO)HW -> NGBOHW
+        out = reshape_dim_outof(2, rhs.sizes()[*rhs_bdim], out);
```
same size nit here

samdow reviewed (approved):
LGTM! Thanks for the clear comments

zou3519 committed:
Update on "[functorch] fix conv_transpose with groups batching rule"


The original batching rule didn't work in all cases, so this PR fixes it.

Test Plan:
 - added new test case to conv_transpose2d's OpInfo. Surprisingly the
 other test cases didn't catch the bug.
 - Removed some xfails and skips as a result.

[ghstack-poisoned]

zou3519 committed:
Update on "[functorch] fix conv_transpose with groups batching rule"


The original batching rule didn't work in all cases, so this PR fixes it.

Test Plan:
 - added new test case to conv_transpose2d's OpInfo. Surprisingly the
 other test cases didn't catch the bug.
 - Removed some xfails and skips as a result.

[ghstack-poisoned]
